src/main/java/teammates/ui/webapi/action/GetSessionResultsAction.java

```diff
@@ -59,12 +60,24 @@ public void checkSpecificAccessControl() {
 
     private StudentAttributes getStudent(String courseId) {
         if (userInfo == null) {
-            String regkey = getNonNullRequestParamValue(Const.ParamsNames.REGKEY);
-            return logic.getStudentForRegistrationKey(regkey);
+            String regkey = getRequestParamValue(Const.ParamsNames.REGKEY);
+            if (StringHelper.isEmpty(regkey)) {
+                throw new UnauthorizedAccessException("Student account is required to access this resource.");
+            } else {
+                return logic.getStudentForRegistrationKey(regkey);
+            }
         }
         return logic.getStudentForGoogleId(courseId, userInfo.id);
     }
 
+    private InstructorAttributes getInstructor(String courseId) {
```

Comment:
This method and the previous one `getStudent` do not indicate in their name that they will throw `UnauthorizedAccessException`, and it doesn't look like it's the right place either.